### PR TITLE
Typosquatting experiment

### PR DESCRIPTION
**Pull request by andrew**
I've added a way for Libraries to detect and record potential typo-squatted names of packages using Levenshtein distance, including a page in the Experiments section that shows recently created packages that match names of similar existing libraries.

screen shot 2017-08-02 at 1 51 18 pm

Inspired by this tweet: https://twitter.com/o_cee/status/892306836199800836 and this follow up blogpost: https://iamakulov.com/notes/npm-malicious-packages/ about malicous typosquatted packages, could be useful for security researchers.

### CODE REVIEW

**Comment by andrew** [comment]
We can backfill the list too as long as we only check names of packages published before each package, seems unlikely that you'd be able to typosquat a package that didn't exist yet.

File: app/views/layouts/_footer.html.erb [hunk 1]
```diff
@@ -45,6 +45,7 @@
           <li><%= link_to 'Bus Factor', bus_factor_path %></li>
           <li><%= link_to 'Digital Infrastructure', digital_infrastructure_path %></li>
           <li><%= link_to 'Unseen Infrastructure', unseen_infrastructure_path %></li>
+          <li><%= link_to 'Typosquatted Libraries', unseen_infrastructure_path %></li>
```
Comment (tswsl1989) [inline app/views/layouts/_footer.html.erb]: Should this be ```similar_names_path```?

Comment (andrew) [inline app/views/layouts/_footer.html.erb]: @tswsl1989 yeah, good spot!

**Comment by andrew** [comment]
Going to close this as I don't think it will ever ship in this form

**Pull request closed without merging**